README.md

```diff
@@ -4,3 +4,105 @@ I am a simple downloader that downloads images from URLs in a CSV and names them
 
 ## Requirements
 Python 3.7+
+
+## Installation
+```bash
+pip install git+https://github.com/Imageomics/cautious-robot
+```
+
+## How it Works
+
+Cautious-robot will check the provided CSV for `IMG_NAME`, `URL`, and `SUBFOLDERS` (if provided), then download all images that have a value in the `IMG_NAME` column. Note that choice of image filename should be unique , especially if working in a flat directory, as the images will otherwise be overwritten. Images that have a filename but no `URL` are recorded in the error log. If desired, a secondary output directory will be created with square copies of the images downsampled to the desired size (e.g., 256 x 256). Parameters such as time to wait between retries, the maximum number of times to try retrying an image, and which index of the CSV to start with can all also be passed. Cautious-robot will retry image downloads when the following responses are returned: `429, 500, 502, 503, 504`.
+
+After downloading the images, cautious-robot calls [`sum-buddy`](https://github.com/Imageomics/sum-buddy) to calculate and record checksums of the `OUTPUT` folder contents.
+
+### Command Line Usage
+```
+usage: cautious-robot [-h] -i [INPUT_FILE] -o [OUTPUT_DIR] [-s [SUBDIR_COL]] [-n [IMG_NAME_COL]] [-u [URL_COL]] [-w WAIT_TIME]
+                      [-r MAX_RETRIES] [-l SIDE_LENGTH] [-x STARTING_IDX] [-a CHECKSUM_ALGORITHM]
+
+options:
+  -h, --help            show this help message and exit
+
+required arguments:
+  -i [INPUT_FILE], --input-file [INPUT_FILE]
+                        path to CSV file with urls.
+  -o [OUTPUT_DIR], --output-dir [OUTPUT_DIR]
+                        main directory to download images into.
+
+optional arguments:
+  -s [SUBDIR_COL], --subdir-col [SUBDIR_COL]
+                        name of column to use for subfolders in image directory (defaults to flat directory if left blank)
+  -n [IMG_NAME_COL], --img-name-col [IMG_NAME_COL]
+                        column to use for image filename (default: filename)
+  -u [URL_COL], --url-col [URL_COL]
+                        column with URLs to download (default: file_url)
+  -w WAIT_TIME, --wait-time WAIT_TIME
+                        time to wait between tries (default: 3)
+  -r MAX_RETRIES, --max-retries MAX_RETRIES
+                        max times to retry download on a single image (default: 5)
+  -l SIDE_LENGTH, --side-length SIDE_LENGTH
+                        number of pixels per side for downsampled images (default: no downsized images created)
```

Comment:
```suggestion
                        number of pixels per side for resized square images (default: no resized images created)
```
Since upscaling is equally possible.